src/main/java/robot/subsystems/gripper/commands/TakeCubeCommand.java

```diff
@@ -0,0 +1,41 @@
+package robot.subsystems.gripper.commands;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.command.Command;
+import robot.Robot;
+
+public class TakeCubeCommand extends Command {
+    private Timer timer = new Timer();
+    private final double timeout = 5;
+
+    public TakeCubeCommand() {
+        requires(Robot.gripperSubsystem);
+    }
+
+    // Called just before this Command runs the first time
+    protected void initialize() {
```

Comment:
You should also reset your timer here.
```suggestion
    protected void initialize() {
    timer.reset();
```

Reply:
initialize happens only once so the timer will be destroyed with the class after you press again the button.

Reply:
`timer` will be destroyed with the object (not class) regardless of whether you `reset` it or not.